Source/ZoomNet/Extensions/Internal.cs

```diff
@@ -500,17 +500,9 @@ internal static JToken GetProperty(this JToken item, string name, bool throwIfMi
 			return property;
 		}
 
-		internal static T GetPropertyValue<T>(this JToken item, string name, T defaultValue = default)
+		internal static T GetPropertyValue<T>(this JToken item, string name, T defaultValue = default, bool throwIfMissing = true)
```

Comment:
Is this change necessary for this PR? I suspect it isn't so I would prefer to exclude it and submit a separate issue+PR for this change so we can have a dialogue about it without getting in the way of merging the "Rooms" PR.

Besides, this change is the reason that you had to modify files such as the oauth handler and the webhook parser which have nothing to do with the "Rooms" resource so you can back out these changes as well and make this PR much simpler and easier for me to approve.